### PR TITLE
Docker operator

### PR DESCRIPTION
**Pull request by asnir**
This PR adds support for running docker containers by adding a
DockerOperator (it's based on this PR https://github.com/airbnb/airflow/pull/733 ). 
It adds the following features support:
- Jinja formatting
- XCom
- Support multiple commands for entrypoint (It might be useful for sealed containers)
- Code examples

### CODE REVIEW

File: airflow/example_dags/docker_copy_data.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
+'''
+This sample "listen to directory". move the new file and print it, using docker-containers.
+The following operators are being used: DockerOperator, BashOperator & ShortCircuitOperator.
+TODO: Review the workflow, change it accordingly to to your environment & enable the code.
+'''
+
+# from __future__ import print_function
```
Comment (criccomini) [inline airflow/example_dags/docker_copy_data.py:7]: Curious why this whole file is commented out?

**Comment by criccomini** [comment]
What's current status on this? Looks like a super useful diff.

**Comment by criccomini** [comment]
Also, does STDOUT get funneled to the log directory, so that Airflow web UI can see it?

File: airflow/example_dags/docker_copy_data.py (hunk 1 above)
Comment (asnir) [inline airflow/example_dags/docker_copy_data.py:7]: @criccomini 
You just need to adapt it to your environment, for example you might change the following settings:
- docker_url - replace it with swarm/docker endpoint
- volumes - '/your/host/input_dir/path' & etc.

_p.s. Race condition may occur (handling the same file again)._

Comment (asnir) [inline airflow/example_dags/docker_copy_data.py:7]: p.s. you have another code example: https://github.com/asnir/airflow/commit/f1063c9dfadd65a4a1b00398f40245679185e4c2

``` python
t3 = DockerOperator(api_version='1.19',
     docker_url='tcp://localhost:2375',
     command='/bin/sleep 30',
     image='centos:latest',
     network_mode='bridge',
     task_id='docker_op_tester',
     dag=dag)

```

**Comment by asnir** [comment]
@criccomini yes, you can view the STDOUT via airflow web-ui. (just try it :-) ).
The code is done and waiting to be accepted.

**Comment by bolkedebruin** [comment]
@mistercrunch quite some work has gone into this any objections?

**Comment by mistercrunch** [comment]
I think this is going to be a popular operator. Thanks everyone!

**Pull request merged**